imap_processing/ialirt/l0/process_hit.py

```diff
@@ -0,0 +1,210 @@
+        "TRIG_IB_EVNT",
+        "NASIDE_IALRT",
+        "NBSIDE_IALRT",
+        *[f"ERATE_{i}" for i in range(1, 6)],
+        "L12A",
+        "L123A",
+        "PENA",
+        "L12B",
+        "L123B",
+        "PENB",
+        "SLOW_RATE_51",
+        "SLOW_RATE_52",
+        "SLOW_RATE_53",
+        "SLOW_RATE_54",
+        "H_06_08",
+        "H_12_15",
+        "H_15_70",
+        "HE4_06_08",
+        "HE4_15_70",
+    ]
+
+
+def find_groups(data: xr.Dataset) -> xr.Dataset:
+    """
+    Find groups in the data.
+
+    Parameters
+    ----------
+    data : xr.Dataset
+        HIT Dataset.
+
+    Returns
+    -------
+    grouped_data : xr.Dataset
+        Grouped data.
+    """
+    subcom_range = (0, 59)
+
+    data = data.sortby("hit_sc_tick", ascending=True)
+
+    # Filter out data before the first subcom=0 and after the last subcom=59
+    start_sc_ticks = data["hit_sc_tick"][(data["hit_subcom"] == subcom_range[0]).values]
+    start_sc_tick = start_sc_ticks.data.min()
+    last_sc_ticks = data["hit_sc_tick"][([data["hit_subcom"] == subcom_range[-1]][-1])]
+    last_sc_tick = last_sc_ticks.data.max()
+
+    grouped_data = data.where(
+        (data["hit_sc_tick"] >= start_sc_tick) & (data["hit_sc_tick"] <= last_sc_tick),
+        drop=True,
+    )
+
+    # Create group labels
+    group_labels = np.searchsorted(
+        start_sc_ticks, grouped_data["hit_sc_tick"], side="right"
+    )
+    grouped_data["group"] = ("group", group_labels)
+
+    return grouped_data
+
+
+def create_l1(
+    fast_rate_1: xr.DataArray,
+    fast_rate_2: xr.DataArray,
+    slow_rate: xr.DataArray,
+) -> dict[str, float]:
+    """
+    Create L1 data dictionary.
+
+    Parameters
+    ----------
+    fast_rate_1 : xr.DataArray
+        Fast rate 1 DataArray.
+    fast_rate_2 : xr.DataArray
+        Fast rate 2 DataArray.
+    slow_rate : xr.DataArray
+        Slow rate DataArray.
+
+    Returns
+    -------
+    l1 : dict
+        Dictionary containing parsed L0 packet data.
+    """
+    fast_rate_1_dict = {
+        prefix: value
+        for prefix, value in zip(HITPrefixes.FAST_RATE_1.value, fast_rate_1.data)
+    }
+    fast_rate_2_dict = {
+        prefix: value
+        for prefix, value in zip(HITPrefixes.FAST_RATE_2.value, fast_rate_2.data)
+    }
+    slow_rate_dict = {
+        prefix: value
+        for prefix, value in zip(HITPrefixes.SLOW_RATE.value, slow_rate.data)
+    }
+
+    l1 = {**fast_rate_1_dict, **fast_rate_2_dict, **slow_rate_dict}
+
+    return l1
+
+
+def process_hit(xarray_data: xr.Dataset) -> list[dict]:
+    """
+    Create L1 data dictionary.
+
+    Parameters
+    ----------
+    xarray_data : dict(xr.Dataset)
+        Dictionary of xarray data including a single
+        set for processing.
+
+    Returns
+    -------
+    hit_data : dict
+        Dictionary final data product.
+    """
+    hit_data = []
+    grouped_data = find_groups(xarray_data)
+
+    for group in np.unique(grouped_data["group"]):
```

Comment:
I think this might be xarray's groupby function: https://docs.xarray.dev/en/stable/user-guide/groupby.html

```suggestion
    for group in grouped_data.groupby("group"):
```

Then you wouldn't have to do the `grouped_data["group"] == group` below, you'd already have the subset entries.